detectors/node/opentelemetry-resource-detector-alibaba-cloud/test/detectors/AlibabaCloudEcsDetector.test.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as nock from 'nock';
+import * as assert from 'assert';
+import { Resource } from '@opentelemetry/resources';
+import { alibabaCloudEcsDetector } from '../../src';
+import {
+  assertCloudResource,
+  assertHostResource,
+} from '@opentelemetry/test-utils';
+
+const ALIYUN_HOST =
+  'http://' + alibabaCloudEcsDetector.ALIBABA_CLOUD_IDMS_ENDPOINT;
```

Comment:
is this really `http` and not `https` ?

Reply:
Yeah, it is `http`. The address is not available in the public network. The IP range is reserved for carrier usage.